packages/text-field/Input.tsx

```diff
@@ -102,7 +102,7 @@ export default class Input<T extends {}> extends React.Component<
     if (handleValueChange && value) {
       handleValueChange(value, () => foundation.setValue(value));
```

Comment:
Because we no longer have the security of the foundation always being set, we need a guard here to check if the foundation exists. We also need a case in the componentDidUpdate to do this same thing if the foundation changes from `undefined` --> `MDCTextFieldFoundation` (no existing --> existing). This also goes for the same 2 lines below:

```
 if (isValid !== undefined && foundation !== null) {
  foundation.setUseNativeValidation(false);
  foundation.setValid(isValid);
    }	 
```

Does that make sense?

Reply:
oh, yeah! I'll try that